### ISSUE TITLE
Problem by connecting MATLAB R2019a & REFPROP 10.0

### ISSUE THREAD

**Issue by osamaaljolani** [usnistgov/REFPROP-wrappers#328]
### Description

I have a problem by connecting MATLAB with REFPROP. The connection was successful before, but since some days my code doesn't work as MATLAB give me always an error that **byref() argument must be a ctypes instance, not 'float'.** 

In addition, in the ctREFPROP.py window it is written there:
**''The function REFPROPdll could not be loaded from the shared library."**  

**Expected behavior:** That the code works normally as it was working before without problems.

**Actual behavior:** When MATLAB connects with REFPROP always I have the following error:
_Error using ctREFPROP>REFPROPdll (line 2500)
Python Error: TypeError: byref() argument must be a ctypes instance, not 'float'_

### Versions

**REFPROP Version:** 10.0
**MATLAB Version:** R2019a
**Operating System and Version:** Windows 10 
**Access Method:** Python 3.7.6 

### Additional Information

Here I give a part of my code as an example such that I have the same error as I have written above.

```matlab
clear all
L2=1;
for k=1:32
    
L2=L2+0.5;
L1(k)=L2;
end

for p=1:32
    
    L=L1(p);
    
% air finned heat exchangers
%1_Heat Exchanger Duty (W)
Q=25382.5;
%2_ _Physical Property Data
% fluids
fluida='air';
fluidc='co2';
% initial temperatures in oC
Tci_c=85.07;
Tci=Tci_c+273.15;
Tco_c=38;
Tco=Tco_c+273.15;
Tai_c=35;
Tai=Tai_c+273.15;
Tao_c=42.001;
Tao=Tao_c+273.15;

% mass fluxes (kg/s)
mc=0.145833; %R744
mw=0.801713;   %Water
Cpw=Q/(mw*(Tao-Tai));

% Mean temperature air side
Ta_meanc=mean([Tao_c Tai_c]);
Ta_mean=Ta_meanc+273.15;
% Mean temperature CO2 side
Tc_meanc=mean([Tco_c Tci_c]);
Tc_mean=Tc_meanc+273.15;

% pressures (Pa)
Pc=9300000;
Pa=100000;

% REFPROP V10 initialization and setup
RP = py.ctREFPROP.ctREFPROP.REFPROPFunctionLibrary('C:\Program Files (x86)\REFPROP'); % initialize REFPROP
RP.SETFLUIDSdll(['C:/Program Files (x86)/REFPROP/FlUIDS/', fluidc, '.fld']); % set fluid
iUnits=RP.GETENUMdll(int8(0),'MASS BASE SI').iEnum; % unit system: MASS BASE SI
iMass=int8(1); % all inputs and outputs are given on a mass basis
iFlag=int8(0); % routine SATSPlN not called
z={1.0}; % composition

% fluid properties CO2 side
%------------inlet----------%
r=RP.REFPROPdll('','TP','H,D,S,CP,VIS,TCX,PRANDTl',iUnits,iMass,iFlag,Tci,Pc,z);
x=r.q; % quality
o=double(r.Output);
hci=o(1); % enthalpy (J/kg)
rci=o(2); % density (kg/m^3)
sci=o(3); % entropy (J/kgK)
cpci=o(4); %specific heat capacity (J/KgK)
mci=o(5); %viscosity (Pa*s)
lci=o(6); %Thermal conductivity W/mK
Prci=o(7); %Prandtl Number

end
```

![MATLAB](https://user-images.githubusercontent.com/71182472/93232122-d0898b80-f779-11ea-9d40-f5eb23662f5a.PNG)
![py](https://user-images.githubusercontent.com/71182472/93232184-e5661f00-f779-11ea-9fa8-346749ceea47.PNG)

**Comment by ianhbell** [comment]
Please provide your complete code example in a code block (within triple backticks) to make it easier for copy-pasting to test the behavior. Also, please reduce your example to the minimal, runnable set of code that we can reproduce the error. What is the type of Tci?

**Comment by osamaaljolani** [comment]
I updated the issue with the example, so with this example I still have the same error. Tci is the inlet temperature of CO2 into the gas cooler.

**Comment by ianhbell** [comment]
Have you tried an int8() on iUnits? I think your problem might be just type conversion in this case.

**Comment by ianhbell** [comment]
What does RP.RPVersion() give you?

**Comment by osamaaljolani** [comment]
Still I didn't try to use int8(), but I will try it now. 

When I give the command RP.RPVersion() it gives me the following:

```
>> RP.RPVersion()

ans = 

  Python str with no properties.

    ******
```

**Comment by ianhbell** [comment]
Ok, that is "good" in that it is clear you have really REFPROP 10.0.0.  If you want a patch that will instead print 10.0.0.2 instead, please email me at ian.bell@nist.gov.

**Comment by osamaaljolani** [comment]
I have installed the files and now my REFPROP is 10.0002, but still I have the same problem as you see in the attached screenshot. Regarding the other point that you remembered about **int8()**, I used it in the iUnits, iMass & iFlag as you see in my example. Did you mean something else?

![10 0002](https://user-images.githubusercontent.com/71182472/93248289-f15cdb80-f78f-11ea-84de-fc3927103845.PNG)

**Comment by ianhbell** [comment]
What about this minimal script (the sort of example I was looking for):

```matlab
root = 'C:/Program Files (x86)/REFPROP';
RP = py.ctREFPROP.ctREFPROP.REFPROPFunctionLibrary(root); % initialize REFPROP
RP.SETPATHdll(root)
iMass=int8(1); % all inputs and outputs are given on a mass basis
iFlag=int8(0); % routine SATSPlN not called
z={1.0}; % composition

Tci = 85.07+273.15;
Pc = 9300000.0;
r = RP.REFPROPdll('CO2','TP','H,D,S,CP,VIS,TCX,PRANDTL',RP.MASS_BASE_SI,iMass,iFlag,Tci,Pc,z);
double(r.Output)
```

**Comment by osamaaljolani** [comment]
Unfortunately it didn't work, still I have the same error. 

![script](https://user-images.githubusercontent.com/71182472/93251952-2a4b7f00-f795-11ea-9e9d-e7b6a7d02b49.PNG)

**Comment by ianhbell** [comment]
Can you restart your MATLAB instance and/or your computer?

**Comment by osamaaljolani** [comment]
I did now but nothing changed. Also today I deinstalled my MATLAB R2019a and reinstalled it again but it didn't work.

If I try to install another version of MATLAB like r2018a or r2018b (as I already have them), do you think this can be something that helps or maybe doesn't worth.

**Comment by ianhbell** [comment]
Worth a shot.  The problem is on the MATLAB side.  I ran this code in python (one-to-one translation)
```python
from ctREFPROP.ctREFPROP import REFPROPFunctionLibrary
root = 'D:/Program Files (x86)/REFPROP';
RP = REFPROPFunctionLibrary(root) # % initialize REFPROP
RP.SETPATHdll(root)
iMass=1 # % all inputs and outputs are given on a mass basis
iFlag=0 # % routine SATSPlN not called
z=[1.0]# % composition

Tci = 85.07+273.15
Pc = 9300000.0
r = RP.REFPROPdll('CO2','TP','H,D,S,CP,VIS,TCX,PRANDTL',RP.MASS_BASE_SI,iMass,iFlag,Tci,Pc,z)
print(r.Output)
```
yielding
```
array('d', [486909.03014168795, 190.43151134477384, 1893.1747510492457, 1627.9930406549145, 2.1226630128365624e-05, 0.031146263032626957, 1.1095008762154075, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0, -9999990.0])
```

**Comment by osamaaljolani** [comment]
I installed R2018b and tried your minimized script, now MATLAB didn't deal with **py** as you see in the attached photo, maybe I need to do something

![Trial](https://user-images.githubusercontent.com/71182472/93258973-68e63700-f79f-11ea-8a15-127c11c6ea96.PNG)
?

**Comment by ianhbell** [comment]
Hmm.. I don't remember how far back MATLAB added the support for Python, but I think it was 2017 or so.  I think they also changed *how* Python support works for one of these versions.  I think it's time for you to talk to Mathworks, I think we've gone as far as we can.

**Comment by osamaaljolani** [comment]
Okay Ian I will let you know when I solve it, thank you for your efforts.

**Comment by osamaaljolani** [comment]
I could solve the problem. It was related to my Python.

**Comment by ianhbell** [comment]
Can you say a little about how you solved the problem?  This issue has popped up a few times now.

**Comment by osamaaljolani** [comment]
Actually I tried many things to solve the problem but nothing worked. Finally I deinstalled my Anaconda and installed it again, then this error didn't appear but the following error appeared instead:

**Undefined variable ''py'' or class ''py.ctREFPROP.ctREFPROP.REFPROPFunctionLibrary''.**

And to solve this error I went back to your topic ''REFPROP for MATLAB'' and when I used the following command the second error was solved:

`[v,e] = pyversion; system(['"', e, '" -m pip install --user -U ctREFPROP'])`

So now my model work very well as it was before. Thank you so much Ian for your time and help.
